### PR TITLE
React hook for manufacturer #287

### PR DESCRIPTION
**Pull request by joshuadkitenge**
## Description

Enter a description of the changes here

## Testing instructions

Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking

connect to {issue number}

### CODE REVIEW

File: src/manufacturer/manufacturerDialog.component.tsx
Comment (joelvdavies) [inline src/manufacturer/manufacturerDialog.component.tsx:1]: When getting a duplicate name error, changing the name doesn't allow save to be clicked.

File: src/manufacturer/manufacturerDialog.component.tsx [hunk 1]
```diff
@@ -10,79 +14,51 @@ import {
...
-  ManufacturerDetails,
+  Manufacturer,
 } from '../app.types';
-import { useAddManufacturer, useEditManufacturer } from '../api/manufacturer';
 import { AxiosError } from 'axios';
+import { useAddManufacturer, useEditManufacturer } from '../api/manufacturer';
 import handleIMS_APIError from '../handleIMS_APIError';
-import { trimStringValues } from '../utils';
+
+const manufacturerSchema = z.object({
```
Comment (joelvdavies) [inline src/manufacturer/manufacturerDialog.component.tsx:27]: The values sent are all empty strings except URL which is undefined. Ideally they would all be undefined or null when not given as the database schema is specifically either null or a value.
image

File: src/manufacturer/manufacturerDialog.component.tsx [hunk 2]
```diff
@@ -10,79 +14,51 @@ import {
...
+  name: z.string().trim().min(1, { message: 'Please enter a name.' }),
+  url: z
+    .string()
+    .url({ message: 'Please enter a valid URL' })
```
Comment (joelvdavies) [inline src/manufacturer/manufacturerDialog.component.tsx:31]: When editing and removing the URL, it gives
image

I would personally do something like
const transformEmptyStringToNull = (value: string) =>
  value === '' ? null : value;

const ZodStringOrNull = z.string().trim().transform(transformEmptyStringToNull);
const ZodUrlOrNull = ZodStringOrNull.transform((value) =>
  z
    .string()
    .url({ message: 'Please enter a valid URL' })
    .nullable()
    .parse(value)
);

So they can all be reused. This for the url also means it does the trim/null comparison first so a url of "    " would also be properly accounted for.

**Review by joelvdavies — COMMENTED** [review]
A few things I noticed

File: src/manufacturer/manufacturerDialog.component.tsx [hunk 3]
```diff
@@ -92,419 +68,269 @@ function ManufacturerDialog(props: ManufacturerDialogProps) {
...
+      ? handleAddManufacturer(data)
+      : handleEditManufacturer(data);
+  };
 
   return (
-    <Dialog open={open} onClose={handleClose} maxWidth="lg" fullWidth>
-      <DialogTitle>{`${
-        type === 'create' ? 'Add' : 'Edit'
-      } Manufacturer`}</DialogTitle>
+    <Dialog open={open} onClose={onClose} maxWidth="lg" fullWidth>
```
Comment (joelvdavies) [inline src/manufacturer/manufacturerDialog.component.tsx:264]:
```suggestion
    <Dialog open={open} onClose={handleClose} maxWidth="lg" fullWidth>
```

File: src/manufacturer/manufacturerDialog.component.tsx [hunk 4]
```diff
@@ -525,25 +351,20 @@ function ManufacturerDialog(props: ManufacturerDialogProps) {
           <Button
             variant="outlined"
             sx={{ width: '50%', mx: 1 }}
-            onClick={handleClose}
+            onClick={onClose}
```
Comment (joelvdavies) [inline src/manufacturer/manufacturerDialog.component.tsx:354]:
```suggestion
            onClick={handleClose}
```

File: src/manufacturer/manufacturerDialog.component.tsx (hunk 2 above)
Comment (joshuadkitenge) [reply src/manufacturer/manufacturerDialog.component.tsx:31]: The issue with this is error message. The error message are tailored to each text field

**Comment by joshuadkitenge** [comment]
#663

**Pull request closed without merging**